### PR TITLE
[1/N] Add Intel GPU Support to Torch Test Cases

### PR DESCRIPTION
**Pull request by daisyden**
For RFC https://github.com/pytorch/pytorch/issues/142029, make the GPU information checkers in inductor general for all the tests:

- Defined GPU_TYPES and get_gpu_type() to torch/_utils.py so that it can be used by both inductor tests and other tests.
- Move the definition of HAS_CUDA/HAS_XPU/HAS_GPU to torch/testing/_internal/common_utils.py, so that it can be shared by inductor tests and other tests.
- Rename the HAS_GPU in inductor to HAS_TRITON_GPU because it also check the availability of triton.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -5550,3 +5550,29 @@ def load_inline(*args, **kwargs):
...
+HAS_XPU = torch.xpu.is_available()
+
+HAS_GPU = HAS_CUDA or HAS_XPU
+
+GPU_TYPE = get_gpu_type()
+
+HAS_MULTIGPU = any(
+    getattr(torch, gpu).is_available() and getattr(torch, gpu).device_count() >= 2
+    for gpu in GPU_TYPES
+)
```
Comment (guangyey) [inline torch/testing/_internal/common_utils.py:5578]: Here needs a blank line.

File: torch/_inductor/utils.py [hunk 2]
```diff
@@ -80,7 +67,7 @@ def get_gpu_type():
 
 from . import config
 from .runtime.runtime_utils import ceildiv as runtime_ceildiv
-
+from torch.testing._internal.common_utils import GPU_TYPES
```
Comment (guangyey) [inline torch/_inductor/utils.py:70]: Here, it doesn't make sense. `inductor` should not depend on `testing` package.

File: torch/_utils.py [hunk 3]
```diff
@@ -1052,3 +1052,13 @@ def fire_callbacks(self, *args: P.args, **kwargs: P.kwargs) -> None:
...
 }
+
+
+GPU_TYPES = ["cuda", "xpu"]
+
+@functools.lru_cache(None)
+def get_gpu_type():
+    avail_gpus = [x for x in GPU_TYPES if getattr(torch, x).is_available()]
+    assert len(avail_gpus) <= 1
+    gpu_type = "cuda" if len(avail_gpus) == 0 else avail_gpus.pop()
```
Comment (PenghuiCheng) [inline torch/_utils.py:1063]: Should it is "no_gpu" if len(avail_gpus) == 0?

File: torch/testing/_internal/common_utils.py [hunk 4]
```diff
@@ -5567,3 +5570,17 @@ def load_inline(*args, **kwargs):
         return func(*args, load_inline=load_inline, **kwargs)
 
     return wrapper
+
+
+HAS_CUDA = torch.cuda.is_available()
```
Comment (PenghuiCheng) [inline torch/testing/_internal/common_utils.py:5575]: In common_device_type.py, there are TEST_CUDA, TEST_XPU...,   could we unify this variable?

File: test/test_utils.py [hunk 5]
```diff
@@ -54,8 +55,6 @@
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests
 
-HAS_CUDA = torch.cuda.is_available()
```
Comment (PenghuiCheng) [inline test/test_utils.py:57]: in common_device_type.py, there is TEST_CUDA , can we unify it ?

**Pull request closed without merging**